addons/crm_iap_lead_enrich/models/res_config_settings.py

```diff
@@ -1,16 +1,24 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import api, models
+from odoo import models, fields
 
 
 class ResConfigSettings(models.TransientModel):
     _inherit = "res.config.settings"
 
-    @api.onchange('lead_enrich_auto')
-    def _onchange_cron_lead_enrich(self):
-        """ change the active status of the cron according to the settings"""
-        if self.module_crm_iap_lead_enrich == True:
-            cron = self.sudo().with_context(active_test=False).env.ref('crm_iap_lead_enrich.ir_cron_lead_enrichment')
-            if cron:
-                cron.active = self.lead_enrich_auto != 'manual'
+    def _get_lead_enrich_auto(self):
+        cron = self.sudo().with_context(active_test=False).env.ref('crm_iap_lead_enrich.ir_cron_lead_enrichment', raise_if_not_found=False)
+        return 'auto' if cron and cron.active else 'manual'
+
+    lead_enrich_auto = fields.Selection(
+        [
+            ('manual', 'Enrich leads on demand only'),
+            ('auto', 'Enrich all leads automatically'),
+        ], string='Enrich lead automatically', required=True,
+        readonly=False, default=_get_lead_enrich_auto, inverse='_inverse_lead_enrich_auto')
```

Comment:
Mmmh I would avoid inverse on res config settings, generally it does not work as intended. I would simply have a computed field and override set_values. I think this is more consistent with the way we use config settings.